CMakeLists.txt

```diff
@@ -0,0 +1,173 @@
+set(PKG_CONFIG_USE_CMAKE_PREFIX_PATH ON)
+list(APPEND CMAKE_PREFIX_PATH "/usr")
+
+# Find Dependencies  ---------------------
+include(cmake/FindLibraries.cmake)
+
+# First find tango if it has not already been found. Returns an interface library
+# called TangoInterfaceLibrary
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+find_package(Tango)
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+
+# Add any include paths from the command line
+set(INCLUDE_PATHS ${INCLUDE_PATHS} ${CMAKE_INCLUDE_PATH})
+
+foreach(INC_PATH ${CMAKE_INCLUDE_PATH})
+   message(STATUS "Extra include path: " ${INC_PATH})
+endforeach()
+
+message(STATUS "Building library paths...")
+
+foreach(LIB_PATH ${CMAKE_LIBRARY_PATH})
+   message(STATUS "Extra library search path: " ${LIB_PATH})
+endforeach()
+
+# Variable to contain a list of all the libs we depend on
+set(HDBPP_MSQL_LIBRARIES
+   libhdb++.so
+   )
+
+# Now load the libraries and check each is found, if not
+# this is an error condition, so halt the cmake config
+foreach(LIB ${HDBPP_MSQL_LIBRARIES})
+   # try the user provided paths first
+   find_library(FOUND_LIB_${LIB} ${LIB} PATHS ${LIBRARY_PATHS} NO_DEFAULT_PATH)
+   # if we could not find it, drop to the system paths
+   if (NOT FOUND_LIB_${LIB})
+      find_library(FOUND_LIB_${LIB} ${LIB})
+   endif()
+
+   if (FOUND_LIB_${LIB})
+      message(STATUS "Found " ${LIB}" at: " ${FOUND_LIB_${LIB}})
+   else()
+      message(FATAL_ERROR "Could not find " ${LIB})
+   endif()
+   list(APPEND LIBS_DIR ${FOUND_LIB_${LIB}})
+endforeach(LIB)
+
+# Source -----------------------------------
+
+add_subdirectory(src)
+
+
+# Build Targets  -----------------------------------
+
+
+# Shared library ------------------------------------------------------
+
+#add_library(shared_library SHARED $<TARGET_OBJECTS:object_library>)
+add_library(libhdbpp_mysql_shared_library SHARED ${SRC_FILES})
+
+target_link_libraries(libhdbpp_mysql_shared_library PRIVATE ${LIBMYSQLCLIENT_LIBRARIES})
+target_link_libraries(libhdbpp_mysql_shared_library PUBLIC ${LIBS_DIR} PRIVATE TangoInterfaceLibrary)
+#target_link_libraries(libhdbpp_mysql_shared_library PRIVATE hdb++)
+
+set_target_properties(libhdbpp_mysql_shared_library PROPERTIES 
+        OUTPUT_NAME hdb++mysql
+        LINK_FLAGS "-Wl,--no-undefined")
+
+# Setting the target includes will allow us to link to this shared
+# library via the cmake project and not require us to include all
+# its dependencies
+target_include_directories(libhdbpp_mysql_shared_library PRIVATE ${LIBMYSQLCLIENT_INCLUDE_DIRS})
+target_include_directories(libhdbpp_mysql_shared_library PRIVATE ${INCLUDE_PATHS})
+target_include_directories(libhdbpp_mysql_shared_library PUBLIC ${CMAKE_CURRENT_LIST_DIR}/src)
+#target_include_directories(libhdbpp_mysql_shared_library PUBLIC "${PROJECT_BINARY_DIR}")
+target_compile_options(libhdbpp_mysql_shared_library PUBLIC -std=c++11 -DRELEASE="${VERSION_STRING}" -Wall)
```

Comment:
In cppTango, we prefer using directives like:
`set(CXX_STANDARD_REQUIRED 11)`
to avoid specifying flags like `-std=c++11` in the compilation options.

See https://github.com/tango-controls/cppTango/pull/732/files#diff-af3b638bc2a3e6c650974192a53c7291

Reply:
This `SET(CXX_STANDARD_REQUIRED 11)` directive is actually already present in this CMakeLists.txt, so it should not be required to add `-std=c++11` in the target_compile_options.

Reply:
I agree. Forgot to remove -std=c++11 after adding SET(CXX_STANDARD_REQUIRED 11)